pagidas/cryptobot
# Initializes a standalone graphql app only responsible to fetch items from rethinkdb

To test
-------
Run `./spin_all.sh build` which wil force build all the containers in docker-compose.
(make sure you `chmod +x` to allow run permissions)

Then, if everything goes smoothly, and by hitting `localhost:8000` you can see messages coming into `messages` table in the rethinkdb. Go to `localhost:9090/graphiql` and try the following query:
```https
{
  messages(mostRecent: 3) {
    productId
    price
    time
  }
}
```
This will order all messages by time in descending, and will fetch 3 of them but pluck only `productId`, `price` and `time`. Of course, you can run the query without the `mostRecent` which will fetch everything. You can ask for whatever fields you want. As expected from the previous implementation of graphql we did in `coinbase-adapter`.

Graphiql is a web client which knows how to do graphql requests, and it's pointing to the service we're spinning. So instead of using postman, you can use that client provided with the app itself.
